### PR TITLE
{AMS} Migrate AMS module to new Microsoft graph

### PR DESCRIPTION
**Pull request by hivyas**
**Related command**
Migrate AMS Module to Microsoft graph according to #22174

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AMS

File: src/azure-cli/azure/cli/command_modules/ams/_params.py [hunk 1]
```diff
@@ -108,12 +108,14 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('account_name', account_name_arg_type)
         c.argument('sp_name', name_arg_type,
                    help="The app name or app URI to associate the RBAC with. If not present, a default name like '{amsaccountname}-access-sp' will be generated.")
+        c.argument('password_display_name', password_arg_type,
+                   help="The friendly name of the password. The actual password will be a strong autogenerated password which will be displayed after the command runs.")
```
Comment (giakas) [inline src/azure-cli/azure/cli/command_modules/ams/_params.py:112]: autogenerated strong password

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py [hunk 2]
```diff
@@ -15,55 +15,50 @@
...
-    app_id = aad_sp.app_id
 
     subscription_id = get_subscription_id(cmd.cli_ctx)
 
-    app_object_id = _get_application_object_id(graph_client.applications, app_id)
-
-    _update_password_credentials(graph_client, app_object_id, sp_password, years)
+    app_object_id = _get_application_object_id(graph_client, app_id)
 
-    _assign_role(cmd, role, sp_oid, ams.id)
```
Comment (giakas) [inline src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:57]: add back

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py [hunk 3]
```diff
@@ -15,55 +15,50 @@
...
-                            cmd.cli_ctx.cloud.endpoints.active_directory,
-                            cmd.cli_ctx.cloud.endpoints.resource_manager, role, xml)
+                            tenant_id, app_id, app_display_name, new_password, password_display_name,
+                            cmd.cli_ctx.cloud.endpoints.management, cmd.cli_ctx.cloud.endpoints.active_directory,
+                            cmd.cli_ctx.cloud.endpoints.resource_manager, '', xml)
 
 
 def create_or_update_assign_sp_to_mediaservice(cmd, client, account_name, resource_group_name, sp_name=None,
-                                               new_sp_name=None, role='Contributor', sp_password=None,
+                                               new_sp_name=None, role=None, password_display_name=None,
```
Comment (giakas) [inline src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:61]: add back.

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py [hunk 4]
```diff
@@ -15,55 +15,50 @@
...
-                                          role='Contributor', sp_password=None, xml=False, years=None):
+                                          password_display_name=None, xml=False, years=1):
     ams = client.get(resource_group_name, account_name)
 
     graph_client = _graph_client_factory(cmd.cli_ctx)
 
     sp_name = _create_sp_name(account_name, sp_name) if sp_name is None else sp_name
-    sp_password = _create_sp_password(sp_password)
 
     app_display_name = sp_name.replace('http://', '')
```
Comment (giakas) [inline src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:35]: double check the http://  behaviour

File: src/azure-cli/azure/cli/command_modules/ams/_params.py [hunk 5]
```diff
@@ -108,12 +108,14 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
-                   help="The password used to log in. Also known as 'Client Secret'. If not present, a random secret will be generated.")
         c.argument('role', help='The role of the service principal.', completer=get_role_definition_name_completion_list)
+        c.argument('password_display_name', password_arg_type,
+                   help="The friendly name of the password. The actual password will be an autogenerated strong password which will be displayed after the command runs.")
+        c.argument('years', help='Number of years for which the secret will be valid. Default: 1 year.', type=int)
         c.argument('xml', action='store_true', help='Enables xml output format.')
-        c.argument('years', help='Number of years for which the secret will be valid. Default: 1 year.', type=int, default=None)
+
+    with self.argument_context('ams account sp create') as c:
+        c.argument('new_sp_name', help="The new app name or app URI to update the RBAC with.")
```
Comment (giakas) [inline src/azure-cli/azure/cli/command_modules/ams/_params.py:118]: What is this for?

Comment (hivyas) [reply src/azure-cli/azure/cli/command_modules/ams/_params.py:118]: If the user wants to change the sp name

Comment (giakas) [reply src/azure-cli/azure/cli/command_modules/ams/_params.py:118]: can they not run create command with new sp name (that will create a new sp right?
)

File: src/azure-cli/azure/cli/command_modules/ams/_client_factory.py [hunk 6]
```diff
@@ -28,14 +28,11 @@ def _auth_client_factory(cli_ctx, scope=None):
 
 def _graph_client_factory(cli_ctx, **_):
     from azure.cli.core._profile import Profile
-    from azure.cli.core.commands.client_factory import configure_common_settings
-    from azure.graphrbac import GraphRbacManagementClient
+    from azure.cli.command_modules.role import graph_client_factory
     profile = Profile(cli_ctx=cli_ctx)
-    cred, _, tenant_id = profile.get_login_credentials(
+    cred, _, tenant_id = profile.get_login_credentials(  # pylint: disable=unused-variable
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/ams/_client_factory.py:33]: Please remove this line as `cred` and `tenant_id` are no longer needed.

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py [hunk 7]
```diff
@@ -300,10 +285,8 @@ def create_application(client, display_name, homepage, years, password,
 
 
 def update_application(client, app_object_id, display_name):
-    app_update_param = ApplicationUpdateParameters(display_name=display_name)
-
     try:
-        return client.patch(app_object_id, app_update_param)
+        return client.application_patch(app_object_id, {"displayName": display_name})
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:289]: I am about to rename it to `application_update`. Will let you know when I finish.

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:289]: This has been done in https://github.com/Azure/azure-cli/pull/22798

**Review by jiasli — APPROVED** [review]

**Pull request merged**